### PR TITLE
Fix edge case in LaminarCurrentSourceDensity, use Poetry for builds, Python >= 3.10

### PR DESCRIPTION
**Pull request by espenhgn**
## Description

Updates LFPykit's packaging and CI to modern Python/PEP 517 tooling while fixing an edge case in `LaminarCurrentSourceDensity` that caused non-zero CSD sums in certain geometries.

**Changes:**
- Fix `LaminarCurrentSourceDensity.get_transformation_matrix()` fraction calculation when radius constraints are non-binding (z-plane crossing vs r–z intersection).
- Add Poetry-based `pyproject.toml`, remove legacy `setup.py` / `setup.cfg`, and bump version to `0.6.0` with Python `>=3.10`.
- Fix `[tool.poetry.extras]`: declare `pytest`, `sympy`, `sphinx`, `numpydoc`, `sphinx_rtd_theme`, and `recommonmark` as optional dependencies in `[tool.poetry.dependencies]` so that `poetry install -E tests` and `poetry install -E docs` work correctly, mirroring the old `extras_require` behavior from `setup.py`.
- Extend CI matrix to newer Python versions and add regression tests covering the edge case fix.

## Motivation and Context

The `LaminarCurrentSourceDensity` class produced non-zero CSD sums in certain geometries due to an incorrect fraction calculation. The legacy `setup.py`-based packaging was replaced with a Poetry/PEP 517 setup for modern tooling support. The initial Poetry extras configuration was missing the required optional dependency declarations, preventing extras from being installable.

## How Has This Been Tested

- Regression tests added covering the `LaminarCurrentSourceDensity` edge case.
- Poetry extras validated by ensuring all packages listed in `[tool.poetry.extras]` are declared as optional in `[tool.poetry.dependencies]`.
- CI matrix extended to newer Python versions to verify compatibility.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

---

✨ Let Copilot coding agent [set things up for you](https://github.com/LFPy/LFPykit/issues/new?title=✨+Set+up+Copilot+instructions&body=Configure%20instructions%20for%20this%20repository%20as%20documented%20in%20%5BBest%20practices%20for%20Copilot%20coding%20agent%20in%20your%20repository%5D%28https://gh.io/copilot-coding-agent-tips%29%2E%0A%0A%3COnboard%20this%20repo%3E&assignees=copilot) — coding agent works faster and does higher quality work when set up for your repo.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+python = ">=3.10"
+numpy = ">=1.15.2"
+scipy = ">=1.5.2"
+meautility = ">=1.5.1"
+
+[tool.poetry.extras]
+tests = ["pytest", "sympy"]
+docs = ["sphinx", "numpydoc", "sphinx_rtd_theme", "recommonmark"]
+
+[tool.poetry.dev-dependencies]
```
[the remark this answers is not in the document]
Comment (espenhgn) [reply pyproject.toml]: @copilot apply changes based on [this feedback](https://github.com/LFPy/LFPykit/pull/197#discussion_r3008734150)

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+[build-system]
+requires = ["poetry-core>=1.0.0"]
+build-backend = "poetry.core.masonry.api"
```
[the remark this answers is not in the document]
Comment (espenhgn) [reply pyproject.toml:3]: @copilot apply changes based on [this feedback](https://github.com/LFPy/LFPykit/pull/197#discussion_r3008734182)

File: lfpykit/models.py [hunk 3]
```diff
@@ -1994,6 +1994,9 @@ def __init__(self, cell, z, r):
         assert r.ndim == 1, 'r.ndim != 1'
         assert r.shape[0] == z.shape[0], 'r.shape[0] != z.shape[0]'
         assert np.all(r > 0), 'r must be greater than 0'
+        len_msg = "All compartments must be shorter than CSD resolution. " +\
+            "Increase cell spatial resolution or decrease CSD resolution."
+        assert np.max(np.diff(z, axis=-1)) > np.max(cell.length), len_msg
```
Comment (espenhgn) [inline lfpykit/models.py]: Maybe throwing a [UserWarning](https://docs.python.org/3/library/warnings.html) would be better, rather than terminating simulations using legacy codes or similar.

Comment (torbjone) [reply lfpykit/models.py]: Agreed. For testing purposes, it can be convenient to be allowed to have long compartments.

**Review by torbjone — APPROVED** [review]
I think this is ready to merge.
